nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,658 @@
+        or have no effect. See scikit-learn documentation for more information.
+
+    screening_percentile: int, float, optional, in the closed interval [0., 100]
+        Perform an univariate feature selection based on the Anova F-value for
+        the input data. A float according to a percentile of the highest scores.
+        Defaults to 20.
+
+    pos_label: str or None
+        The positive class label in case of a binary classification. It is
+        used in the case of binary classification, and only the following
+        scoring metrics require it: f1, precision, and recall.
+
+    scoring : string or callable, optional
+        The scoring strategy to use. See the scikit-learn documentation
+        If callable, takes as arguments the fitted estimator, the
+        test data (X_test) and the test target (y_test) if y is
+        not None.
+
+    smoothing_fwhm: float, optional
+        If smoothing_fwhm is not None, it gives the size in millimeters of the
+        spatial smoothing to apply to the signal.
+
+    standardize : boolean, optional
+        If standardize is True, the time-series are centered and normed:
+        their variance is put to 1 in the time dimension.
+
+    target_affine: 3x3 or 4x4 matrix, optional
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    target_shape: 3-tuple of integers, optional
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    mask_strategy: {'background' or 'epi'}, optional
+        The strategy used to compute the mask: use 'background' if your
+        images present a clear homogeneous background, and 'epi' if they
+        are raw EPI images. Depending on this value, the mask will be
+        computed from masking.compute_background_mask or
+        masking.compute_epi_mask. Default is 'background'.
+
+    memory: instance of joblib.Memory or string
+        Used to cache the masking process.
+        By default, no caching is done. If a string is given, it is the
+        path to the caching directory.
+
+    memory_level: integer, optional
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    n_jobs : int, optional. Default is -1.
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs'.
+
+    verbose : int, optional
+        Verbosity level. Default is False
+    """
+    # XXX Too many arguments (17/5)
+    # XXX Too many local variables (17/15)
+    def __init__(self, estimator='svc', mask=None, cv=None, param_grid=None,
+                 screening_percentile=20, pos_label=None, scoring=None,
+                 smoothing_fwhm=None, standardize=True, target_affine=None,
+                 target_shape=None, mask_strategy='background',
+                 memory=Memory(cachedir=None), memory_level=0,
```

Comment:
I would put `memory=None` and set it within the function to `Memory(cachedir=None)` -- there is always a risk in putting mutable objects as function defaults.